rotkehlchen/chain/bitcoin/xpub.py

```diff
@@ -227,8 +230,26 @@ def _derive_xpub_addresses(
                 self.chain_manager.add_blockchain_accounts(
                     blockchain=blockchain,
                     accounts=new_addresses,
-                    already_queried_balances=new_balances,
                 )
+                if blockchain == SupportedBlockchain.BITCOIN:
+                    balances = self.chain_manager.balances.btc
+                    totals = self.chain_manager.totals.assets[A_BTC]
+                    asset_usd_price = Inquirer.find_usd_price(A_BTC)
+                elif blockchain == SupportedBlockchain.BITCOIN_CASH:
+                    balances = self.chain_manager.balances.bch
+                    totals = self.chain_manager.totals.assets[A_BCH]
+                    asset_usd_price = Inquirer.find_usd_price(A_BCH)
+                else:
+                    raise AssertionError('Should not happen')
+
+                for addr, raw_balance in zip(new_addresses, new_balances):
+                    new_balance = Balance(
+                        amount=raw_balance,
+                        usd_value=raw_balance * asset_usd_price,
+                    )
+                    balances[addr] = new_balance
+                    totals += new_balance
```

Comment:
This will not work. how come the tests don't catch this? This code will need to be tested.

You are appending `new_balance` on the totals in each iteration but that is not saved anywhere. `totals` here is just a reference. So it's not actually saved in `self.chain_manager.totals.assets[A_BTC]`. You should modify that instead.